realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -76,6 +76,12 @@
     // Thread pool for all async operations (Query & transaction)
     static final RealmThreadPoolExecutor asyncTaskExecutor = RealmThreadPoolExecutor.newDefaultExecutor();
 
+    /**
+     * Thread pool executor used for write operations - only one thread is needed as writes cannot
+     * be parallelized.
+     */
+    public static final RealmThreadPoolExecutor writeExecutor = RealmThreadPoolExecutor.newSingleThreadExecutor();
```

Comment:
We should use `WRITE_EXECUTOR` since this is public. If we introduce this, we should probably also redirect `executeTransactionAsync` to use it.

Reply:
But it is fine to create an issue for that and do it in a followup PR